test/functional/test_framework/test_node.py

```diff
@@ -480,7 +480,9 @@ def add_p2p_connection(self, p2p_conn, *, wait_for_verack=True, **kwargs):
             #
             # So syncing here is redundant when we only want to send a message, but the cost is low (a few milliseconds)
             # in comparision to the upside of making tests less fragile and unexpected intermittent errors less likely.
-            p2p_conn.sync_with_ping()
+            # Connections may opt out by passing `sync_with_ping=False`.
+            if sync_with_ping:
+                p2p_conn.sync_with_ping()
```

Comment:
Wouldn't it be confusing to have `add_p2p_connection(..., wait_for_verack=False, sync_with_ping=True)` **not** execute `sync_with_ping()`?

Reply:
Good call. Updated.

Reply:
I see your point, but I don't see when we would want a connection like that and AFAIK the test suite doesn't currently have a use for it.

Reply:
Nit: As a follow-up to @vasild 's comment, would it make sense to consider one of the following, to prevent possible confusion in the future:

- Raise an assertion error if `wait_for_verack=False, sync_with_ping=True`
- Instead of passing two boolean parameters, pass a single parameter like `sync_behavior` which can take in three values `NONE`, `VERACK_ONLY`, `VERACK_AND_PING`?